pages/api/savedInsulinData/index.js

```diff
@@ -0,0 +1,22 @@
+import dbConnect from '../../../dbConnect';
+import SavedInsulinData from '../../../models/InsulinApp';
+import { getAllCategories } from '../../../services/insulinAppDataService';
+
+export default async function handler(request, response) {
+  switch (request.method) {
+    case 'GET':
+      const savedInsulinData = await getAllCategories();
+      return response.status(200).json({ savedInsulinData });
+
+    case 'DELETE':
+      await dbConnect();
+
+      const data = request.body;
+
+      await SavedInsulinData.findOneAndDelete();
```

Comment:
you declare the variable `data` but don't use it anywhere, why?

how does your backend know which item to delete? 

you have some better logic and another delete handler in your dynamic api route `[id].js`. You can delete the whole DELETE case.